receiver/otlpreceiver/otlp.go

```diff
@@ -163,6 +166,9 @@ func (r *otlpReceiver) startProtocolServers(host component.Host) error {
 // Start runs the trace receiver on the gRPC server. Currently
 // it also enables the metrics receiver too.
 func (r *otlpReceiver) Start(_ context.Context, host component.Host) error {
+	onceLogLocalHost.Do(func() {
+		component.LogAboutUseLocalHostAsDefault(r.settings.Logger)
```

Comment:
No big deal, but I think I would log it once per receiver instance instead of once per receiver type. The reason is that the r.settings.Logger has the component name in it, which is specific to the receiver instance.

Reply:
Alternative: if `component.LogAboutUseLocalHostAsDefault` is always logging a message anyway (see my other comment there), then the sync.Once should be at the feature flag level, and not per component?